### ISSUE TITLE
[Crash]: Assert when clicking on Industry Chain button

### ISSUE THREAD

**Issue by 2TallTyler** [OpenTTD/OpenTTD#12057]
### Version of OpenTTD

14.0-beta3

### Steps to reproduce

![industry_chain_crash](https://github.com/OpenTTD/OpenTTD/assets/55058389/cf2e3b09-fe9a-4fce-bfcd-d9e61c65776e)

This is a tricky crash to reproduce. Even in the attached save, it only crashes for some industries and not others of the same type.

It does not seem to be connected to the industry ID, because in the crash save I get the assertion for one Steel Mill (but not the other) as well as a Building Materials Factory.

It also happens when selecting the industry chain from the Industry Chain GUI and the Select Industry dropdown.

1. Play a game with Improved Town Industries 2.9
2. Find a Steel Mill or Building Materials Factory
3. Click on the industry, then on the Industry Chain button
4. Maybe get an assertion failure.

Here is the Steel Mill [NML source](https://github.com/2TallTyler/improved_town_industries/blob/master/src/steel_mill.nml).

### Upload crash files

[crash20240211131433.zip](https://github.com/OpenTTD/OpenTTD/files/14232483/crash20240211131433.zip)

**Comment by glx22** [comment]
```
 	[Cadre en ligne] openttd.exe!google_breakpad::ExceptionHandler::WriteMinidump() Ligne 747	C++
 	openttd.exe!google_breakpad::ExceptionHandler::WriteMinidump(const std::wstring & dump_path, bool(*)(const wchar_t *, const wchar_t *, void *, _EXCEPTION_POINTERS *, MDRawAssertionInfo *, bool) callback, void * callback_context, _MINIDUMP_TYPE dump_type) Ligne 773	C++
 	openttd.exe!CrashLogWindows::WriteCrashDump() Ligne 113	C++
 	[Cadre en ligne] openttd.exe!std::_Func_class<bool>::operator()() Ligne 883	C++
 	openttd.exe!CrashLogWindows::TryExecute(std::basic_string_view<char,std::char_traits<char>> section_name, std::function<bool __cdecl(void)> && func) Ligne 124	C++
 	openttd.exe!CrashLog::MakeCrashLog() Ligne 290	C++
 	openttd.exe!ExceptionHandler(_EXCEPTION_POINTERS * ep) Ligne 341	C++
 	[Code externe]	
 	openttd.exe!raise(int signum) Ligne 547	C++
 	openttd.exe!abort() Ligne 71	C++
 	openttd.exe!FatalErrorI(const std::string & str) Ligne 143	C++
 	openttd.exe!AssertFailedError(int line, const char * file, const char * expression) Ligne 20	C++
>	openttd.exe!CargoesField::ConnectCargo(unsigned char cargo, bool producer) Ligne 2053	C++
 	openttd.exe!CargoesRow::ConnectIndustryAccepted(int column) Ligne 2482	C++
 	[Cadre en ligne] openttd.exe!IndustryCargoesWindow::PlaceIndustry(int) Ligne 2796	C++
 	openttd.exe!IndustryCargoesWindow::ComputeIndustryDisplay(unsigned char displayed_it) Ligne 2870	C++
 	[Cadre en ligne] openttd.exe!IndustryCargoesWindow::OnInvalidateData(int) Ligne 2973	C++
 	[Cadre en ligne] openttd.exe!IndustryCargoesWindow::{ctor}(int) Ligne 2552	C++
 	openttd.exe!ShowIndustryCargoesWindow(unsigned char id) Ligne 3218	C++
 	openttd.exe!IndustryViewWindow::OnClick(Point pt, int widget, int click_count) Ligne 1093	C++
 	openttd.exe!DispatchLeftClickEvent(Window * w, int x, int y, int click_count) Ligne 733	C++
 	openttd.exe!MouseLoop(MouseClick click, int mousewheel) Ligne 2840	C++
 	openttd.exe!HandleMouseEvents() Ligne 2943	C++
 	openttd.exe!WndProcGdi(HWND__ * hwnd, unsigned int msg, unsigned __int64 wParam, __int64 lParam) Ligne 445	C++
 	[Code externe]	
 	openttd.exe!VideoDriver_Win32Base::PollEvent() Ligne 884	C++
 	openttd.exe!VideoDriver::Tick() Ligne 135	C++
 	openttd.exe!VideoDriver_Win32Base::MainLoop() Ligne 895	C++
 	openttd.exe!openttd_main(int argc, char * * argv) Ligne 827	C++
 	openttd.exe!WinMain(HINSTANCE__ * __formal, HINSTANCE__ * __formal, char * __formal, int __formal) Ligne 85	C++
 	[Code externe]	
```

**Comment by PeterN** [comment]
Fixed by #12053.
